File: Examples/OutputProcessing/swistrack-filter.py
#! /usr/bin/env python

import re

# Prepare regular expressions
re_beginframe = re.compile('\$BEGINFRAME,([\d\-]*)\*')
re_particle = re.compile('\$PARTICLE,([\d-]*),([\d\.]*),([\d\.]*),([\d\.]*)\*')

# Read from STDIN continously
framenumber = 0
while True:
	line = raw_input()

	result = re_beginframe.search(line)
	if result != None:
		# This is the beginning of a frame - just store the frame number (first result of regular expression)
		framenumber=result.group(1)
		continue

	result = re_particle.search(line)
	if result != None:
		# This is a particle
		print framenumber, ' ', result.group(1), ' ', result.group(2), ' ', result.group(3), ' ', result.group(4), "\n";
		continue

File: Examples/OutputProcessing/swistrack-filter.pl
#! /usr/bin/perl
use strict;
$|=1;

# Read from STDIN continously
while (my $line=<>) {
	if ($line =~ /\$STEP_START\*/) {
		# This is the beginning of a frame
		print 'We\'ve got a new frame ...', "\n";
	} elsif ($line =~ /\$FRAMENUMBER,([\d-]*)\*/) {
		# This is the frame number (first result of regular expression)
		print '  Framenumber =', $1, "\n";
	} elsif ($line =~ /\$PARTICLE,([\d-]*),([\d\.-]*),([\d\.-]*),([\d\.-]*)\*/) {
		# This is a particle
		print '  Particle: ID = ', $1, ' X = ', $2, ' Y = ', $3, ' alpha = ', $4, "\n";
	} 
}
